### PR TITLE
Feature Requst - Change active lights "off the field" at night

### PR DESCRIPTION
**Pull request by jtsage**
## Why this setting is being suggested:

A fair few of the newer truck and semi mods available are using the existing light system in non-standard ways to provide more "pretty" decoration.  Of note, the RoadRunner+ (modhub) uses LightType=0 (supposed to be headlights) as parking lights, and LightType=2 (supposed to be front works) as the actual headlights.  Similarly, the NMC Reaper Pack (modhub) uses headlights (type 0) as decoration, and the rear worklights (type 1) as the real headlights.  In both of these cases, the lightType I defined as the "real headlights" is the lightType that contains the &lt;realLights> of the mod. Of course, when AutoDrive currently turns on what should be the headlights, the truck proceeds to drive around mostly in the dark.

## What I Did:

Added a setting called "Night Driving Lights" that allows the user, on a per-vehicle basis, to choose which set of lights to illuminate.  I used the "proper" names of the lights, because there is no way of knowing what the mod author actually did.

settingscreen

## What and How I tested:

I used a number of vanilla vehicles, along with a few modded vehicles I knew are currently "broken" and ran a few in-game days.  Some screenshots below...

onfield-nochange
__"Broken" Tractor, on the field.  No change to existing behavior, _aiLightsTypeMask_ is used for a vehicle working in the field__

moved-offfield-overridden
__Same "Broken" Tractor, having just moved off the field, with the new setting on "_Headlights + Front Worklights_"__
a/n: Yes, the parking lights should probably be off for road driving, which would be the _"Front Worklights"_ setting, but they are really pretty.

offfield-standardheadlights
__Vanilla tractor, off the field, with no setting change - behavior is what it has always been__

offfield-overridenstandard1
__Vanilla Semi, off the field, with the setting overridden to "_Headlights + Front Worklights_" to show possibilities__

## Settings available:

 * "No Lights"
 * "Headlights" - the current setting used
 * "Back Worklights"
 * "Headlights + Back Worklights"
 * "Front Worklights"
 * "Headlights + Front Worklights"
 * "Front + Back Worklights"
 * "All Lights" - the equivalent of _aiLightsTypesMask_

## What doesn't work quite right

So, being that I am unfamiliar with all the ins and outs of how AutoDrive works under the hood, I have notice that changing the setting does not take effect until you switch vehicles (or get in and get out).  I would guess this is because of when the per-vehicle settings are read, or perhaps I am reading the setting in scripts/Specialization.lua incorrectly.

Additionally, I wasn't sure exactly how i18n is handled by AutoDrive, so I did add the english version of all of these terms to each of the translation files - if this is the improper method, let me know and I can back those changes out of the commit.  My apologies, as I am not even bi-lingual, and can be of no help in the translation.

### Other bits I didn't add, but thought about

I made a short attempt at a "Daytime Running Lights" option too, but I was unsuccessful - is updateAILights not called if the vehicle is started in the daytime? (It is also entirely possible that I simply mistyped something)

I would also be interested in a "use Hazard Lights" for road driving - at least where I am in the US, it wouldn't be odd to see an oversized load with both beacons and hazards running - the giant seed drill / air carts come to mind (My JD 9620RX with 1870 seed drill, and c850 air cart would look fantastic running down the road with the beacons and hazards blaring)

### CODE REVIEW

**Comment by Axel32019** [comment]
Hello @jtsage,
First of all thanks for spending time and effort to contribute to this project!
The topic of lights is a little more complex than you may have figured out :-)
If you have a look in the list of feature requests you will find some ideas AD should cover regarding lights.
Some of them:
- time control switch lights on/off, independend of map settings - see some lines above your changes
- situations, configurations, drive direction, curve etc.
- on/off street, field

issues: 1549 1704
But now back to your PR:
The LightsTypesMask is defined by Giants and not a linear numbering for the activateable lights.
You may have to take a look in the XML of some base vehicles of these definitions and the GDN: https://gdn.giants-software.com/documentation_scripting_fs19.php?version=script&category=32&class=368
So from the first look at your implementation it is not really a surprise that not all is working as you expect it.
From GDN you may find out how and when the function updateAILights is called.
Please do not understand this as critic, but inspiration...
To summarize: It is highly wellcome if you want to dive into that topic!
Best Regards Axel

**Comment by jtsage** [comment]
Thanks for the further info - and yes, I am interested in diving into this a bit more. Is this the best venue for further discussion, or would something else be better? 

I did find the bit on GDN that is actually doing the bit masking that I basically brute-forced to decimal (which likely will work just fine as long as it's a vehicle that has lightTypes 0, 1, & 2 defined.  I'd bet good money that the results are going to be unpredictable, or if Giants didn't throw in enough error trapping, downright bad for a mod or vanilla vehicle that doesn't.  I'll look much closer at this in the next few days

Of course, I also ignored high beams, at least for now.  I think the biggest guidance I'd be interested in is which of the feature requests are of most interest, and also how to avoid adding just a ton of new settings - right now AD is honestly one of the easier "advanced" mods to understand, likely by design (well done!), and I would hate to have any negative impact on that.

Thanks ~j

**Comment by Axel32019** [comment]
Hello @jtsage , I suggest to collect features here #1820
Please add via comment what is missing, so we can have it together. Next we have to check how to get it in the UI, to not make it to complex for users. For detailed discussion we might also use our discord.
What do you think about this approach?

**Comment by jtsage** [comment]
Sounds good to me.  I'll spend some time reading it over, and adding some of the other stuff I found which may make life a lot easier moving forward.  Fwiw, I am in the discord, same username as here.  Thanks!

**Comment by Axel32019** [comment]
Will be followed-up in #1820
